core-bundle/src/Security/Authentication/AuthenticationSuccessHandler.php

```diff
@@ -73,17 +73,17 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token)
 
         $user = $token->getUser();
 
-        $response = $this->httpUtils->createRedirectResponse($request, $this->determineTargetUrl($request));
-
         if (!$user instanceof User) {
-            return $response;
+            return $this->httpUtils->createRedirectResponse($request, $this->determineTargetUrl($request));
         }
 
         $this->user = $user;
         $this->user->lastLogin = $this->user->currentLogin;
         $this->user->currentLogin = time();
         $this->user->save();
 
+        $response = $this->httpUtils->createRedirectResponse($request, $this->determineTargetUrl($request));
```

Comment:
Can you explain this change? As far as I can see, you just duplicated the code?

Reply:
See https://github.com/contao/contao/commit/259e319790dc0ea454c12ae1373aa46b53b6d21b.
My proposed fix was https://github.com/contao/contao/pull/363/commits/30cb28735cac80566061bd57606d65f4a0b84aae.

Reply:
How is that realted to `createRedirectResponse`?

Reply:
Because `determineTargetUrl()` uses `$this->user` and this could be possibly a `string` value and phpstan does not like that.

Reply:
I don't like this fix, I would have fixed it with the annotation.

Reply:
ah, I see 👍